packages/lodestar-cli/test/unit/lodestar/account.test.ts

```diff
@@ -0,0 +1,114 @@
+import { TestnetName } from "../../../src/testnets";
+import { medallaConfig } from "../../../src/testnets/medalla";
+import { existsSync, readFileSync, readdirSync, unlinkSync } from "fs";
+
+describe.only("account cli", function() {
+  this.timeout("10 min");
+
+  const spy = stub(console, 'log');
+
+  const testnetName = "medalla";
+  const initDefaults = {
+    rootDir,
+    preset: "mainnet",
+    testnet: testnetName as TestnetName,
+    paramsFile: `${rootDir}/config.yaml`,
+    params: {
+      "DEPOSIT_CHAIN_ID": 5,
+      "DEPOSIT_NETWORK_ID": 5
+    },
+  }
+
+  const walletsDir = `${rootDir}/wallets/`;
+  const walletName = "primary";
+
+  const accountDefaults = {
+    ...initDefaults,
+    name: walletName,
+    passphraseFile: "primary.pass",
+  };
+
+  it("should init beacon configuration with --testnet option", async function() {
+    await new Promise(resolve => yargs()
+      .default(initDefaults)
+      .command(init).help().parse(["init"], resolve));
+    await new Promise(resolve => setTimeout(resolve, 6000));
+    expect(existsSync(rootDir)).to.be.true;
+    const beaconConfigPath = `${rootDir}/beacon.config.json`;
+    expect(existsSync(beaconConfigPath)).to.be.true;
+    
+    const beaconConfig = JSON.parse(readFileSync(beaconConfigPath, "utf8"));
+    let templateConfigCopy = JSON.parse(JSON.stringify(medallaConfig));
+    templateConfigCopy = {
+      ...templateConfigCopy,
+      beaconDir: rootDir,
+      configFile: beaconConfigPath,
+      dbDir: `${rootDir}/chain-db`,
+      enrFile: `${rootDir}/enr.json`,
+      eth1: { ...templateConfigCopy.eth1, enabled: false },
+      genesisStateFile: `${rootDir}/genesis.ssz`,
+      peerIdFile: `${rootDir}/peer-id.json`
+    }
+    expect(beaconConfig).to.deep.equal(templateConfigCopy);
+  });
+  
+  it("should create a wallet", async function() {
+    expect(existsSync(rootDir)).to.be.true;
+    await new Promise(resolve => yargs().default(accountDefaults)
+      .command(account).help().parse(["account", "wallet", "create"], resolve));
+    await new Promise(resolve => setTimeout(resolve, 500));
+    expect(existsSync(walletsDir)).to.be.true;
+    const wallets = readdirSync(walletsDir);
+    expect(wallets.length > 0).to.be.true;
+  });
+
+  it("should list existing wallets", async function() {
+    await new Promise(resolve => yargs().default(accountDefaults)
+      .command(account).help().parse(["account", "wallet", "list"], resolve));
+    await new Promise(resolve => setTimeout(resolve, 500));
+    expect(spy.calledWith(walletName)).to.be.true;
+  });
+
+  it("should create new validator", async function() {
+    const wallets = readdirSync(walletsDir);
+    expect(wallets.length > 0).to.be.true;
+    await new Promise(resolve => yargs().default(accountDefaults)
+      .command(account).help().parse(["account", "validator", "create"], resolve));
+    await new Promise(resolve => setTimeout(resolve, 500));
+    expect(spy.calledWith(match.string)).to.be.true;
```

Comment:
It should assert that the validator was created by checking the expected files are there or by some other means. Matching a console.log output seems too brittle.

Reply:
fixed